pgmanage/app/static/pgmanage_frontend/src/components/FileManager.vue

```diff
@@ -399,8 +402,39 @@ export default {
           showToast("error", error.response.data.data);
         });
     },
-    //TODO: implement upload functionality
-    onUpload() {},
+    async handleFileUpload(event) {
```

Comment:
Consider using onUploadProgress and somehow indicate the progress in the UI. For larger file uploads to the remote server this may take more than several seconds and it should be clear for the user that the upload is really happening.